### PR TITLE
Handle nicely cases in which output of the process is not in utf-8

### PR DESCRIPTION
**Pull request by Nikokrock**
When using output=PIPE, self.out will contains either a decoded utf-8
string or a string representation of the bytes produced by the process.
This ensure e3.os.process does not crash in case the output is not in
utf-8. Do likewise for errror.

### CODE REVIEW

File: src/e3/os/process.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import e3.env
```
[the remark this answers is not in the document]
Comment (Nikokrock) [reply src/e3/os/process.py:21]: Fixed

File: src/e3/os/process.py [hunk 2]
```diff
@@ -389,7 +392,7 @@ def add_interpreter_command(cmd_line: CmdLine) -> CmdLine:
                     "stderr": self.error_file.fd,
                     "cwd": cwd,
                     "env": env,
-                    "universal_newlines": True,
+                    "universal_newlines": False,
```
[the remark this answers is not in the document]
Comment (Nikokrock) [reply src/e3/os/process.py:395]: I would prefer to keep it explicit. But I can remove it if you think it's better

File: src/e3/os/process.py [hunk 3]
```diff
@@ -217,8 +219,9 @@ class Run:
         the process has exited, its initial value is None.  When a problem
         running the command is detected and a process does not get
         created, its value gets set to the special value 127.
-    :ivar out: process standard output  (if instanciated with output = PIPE)
-    :ivar err: same as out but for standard error
+    :ivar raw_out: process standard output as bytes (if instanciated with
+        output = PIPE). Use self.out to get a decoded string
+    :ivar raw_err: same as out but for standard error
```
[the remark this answers is not in the document]
Comment (Nikokrock) [reply src/e3/os/process.py]: fixed

File: src/e3/os/process.py [hunk 4]
```diff
@@ -459,6 +460,26 @@ def add_interpreter_command(cmd_line: CmdLine) -> CmdLine:
...
+    @property
+    def err(self) -> str:
+        """Process error as string.
+
+        Attempt is done to decode as utf-8 the output. If the output is not in
+        utf-8 a string representation will be returned
+        (see e3.text.bytes_as_str).
+        """
+        return bytes_as_str(self.raw_err)
+
```
[the remark this answers is not in the document]
Comment (Nikokrock) [reply src/e3/os/process.py:482]: I have fixed the test. Gnatpython was preserving CR in test output. Some testsuite require to keep them too. I think that's a better default

[the remark this answers is not in the document]
Comment (enzbang) [reply src/e3/os/process.py:482]: Please document this change of behavior in the NEWS.md file

File: src/e3/os/process.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (enzbang) [reply src/e3/os/process.py:395]: Given that's a change in the API I think we can keep it explicit.

**Comment by camilo1729** [comment]
Looks good. I dont want to be too picky but  probably a test for not uft-8 output is missing

**Comment by camilo1729** [comment]
Nice thanks for the test. It looks good

**Pull request merged**